agent/testflinger_agent/agent.py

```diff
@@ -243,25 +254,87 @@ def process_jobs(self):
                         == JobState.CANCELLED
                     ):
                         logger.info("Job cancellation was requested, exiting.")
+                        self.add_event_and_post_status_update(
+                            job_queue,
+                            webhook,
+                            events,
+                            TestEvent.CANCELLED,
+                            datetime.now(timezone.utc).isoformat(),
+                            "",
+                        )
                         break
+
                     self.client.post_job_state(job.job_id, phase)
                     self.set_agent_state(phase)
-                    exitcode = job.run_test_phase(phase, rundir)
-
+                    self.add_event_and_post_status_update(
+                        job_queue,
+                        webhook,
+                        events,
+                        TestEvent("started_" + phase),
+                        datetime.now(timezone.utc).isoformat(),
+                        "",
```

Comment:
Why is it an empty string? We can probably set the default value to be “” instead of requiring to pass it

Reply:
Changed to default to ""